apps/full-site-editing/full-site-editing-plugin/block-inserter-modifications/index.php

```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * Block Inserter Modifications
+ *
+ * @package A8C\FSE
+ */
+
+namespace A8C\FSE\BLOCK_INSERTER_MODIFICATIONS;
+
+/**
+ * Enqueues a submodule script by its filename.
+ *
+ * @param string  $filename Name of the script file w/o extension.
+ * @param boolean $in_footer Whether to enqueue the script before </body> instead of in the <head> (optional).
+ */
+function enqueue_script( $filename, $in_footer = false ) {
+	$asset_file   = __DIR__ . '/dist/' . $filename . '.asset.php';
+	$asset        = file_exists( $asset_file ) ? require_once $asset_file : null;
+	$dependencies = isset( $asset['dependencies'] ) ? $asset['dependencies'] : array();
+	$version      = isset( $asset['version'] ) ? $asset['version'] : filemtime( __DIR__ . '/' . $filename . '.js' );
```

Comment:
Do we expect instances where we have a script file but no associated assets file? If not, should we throw an error here, [like in premium content](https://github.com/Automattic/wp-calypso/blob/master/apps/full-site-editing/full-site-editing-plugin/premium-content/premium-content.php#L44-L48)?

Reply:
> Do we expect instances where we have a script file but no associated assets file?

It should never happen. Even scripts with no dependencies still get the asset file generated with the file hash and an empty dependencies array.

Let's consider asset file required

Reply:
Requirement added in https://github.com/Automattic/wp-calypso/pull/41822/commits/56f4e29afd8d1ab50d55a127ea5b5445a627f841

Reply:
With the file now being required, can the fallbacks here be removed and dependency & version moved inline with the `wp_enqueue_script()`?

Reply:
Updated in https://github.com/Automattic/wp-calypso/pull/41822/commits/c4c1e40f5ca360cc72729fd683717281c13440d2